nautobot/utilities/testing/filters.py

```diff
@@ -6,9 +9,49 @@
 
 @tag("unit")
 class FilterTestCases:
-    class FilterTestCase(TestCase):
+    class BaseFilterTestCase(TestCase):
         """Base class for testing of FilterSets."""
 
+        def get_filterset_test_values(self, field_name, queryset=None):
+            """Returns a list of distinct values from the requested queryset field to use in filterset tests.
+
+            Returns a list for use in testing multiple choice filters. The size of the returned list is random
+            but will contain at minimum 2 unique values. The list of values will match at least 2 instances when
+            passed to the queryset's filter(field_name__in=[]) method but will fail to match at least one instance.
+
+            Args:
+                field_name: The name of the field to retrieve test values from.
+                queryset: The queryset to retrieve test values. Defaults to `self.queryset`.
+
+            Returns:
+                list: A list of unique values derived from the queryset.
+
+            Raises:
+                ValueError: Raised if unable to find a combination of 2 or more unique values
+                    to filter the queryset to a subset of the total instances.
+            """
+            test_values = []
+            if queryset is None:
+                queryset = self.queryset
+            qs_count = queryset.count()
+            values_with_count = queryset.values(field_name).annotate(count=Count(field_name)).order_by("count")
+            for value in values_with_count:
+                # randomly break out of loop after 2 values have been selected
+                if len(test_values) > 1 and random.choice([True, False]):
+                    break
+                if value["count"] < qs_count:
```

Comment:
Correct me if I'm wrong. Would `value["count"]` ever be greater than `qs_count`? And we do not append the `value[field_name]` when `qs_count == value["count"]`. Does that ignore the last eligibe value in `values_with_count`?

Reply:
> Correct me if I'm wrong. Would value["count"] ever be greater than qs_count? 

`value["count"]` would never be greater than `qs_count` because the former is a subset of the latter. The max it could ever be is `qs_count`.


> And we do not append the value[field_name] when qs_count == value["count"]. Does that ignore the last eligibe value in values_with_count?

That's correct and it's by design. The condition `if value["count"] >= qs_count` will only ever match on the final object in the loop and we always want to test a filter on a subset of objects in the queryset. When a filter fails it returns the original queryset. If our test expects a filter to return all of the objects in the queryset we don't know the difference between success and failure.